lib/backend/dynamo/dynamodbbk.go

```diff
@@ -332,8 +332,35 @@ func (b *DynamoDBBackend) fullPath(bucket ...string) string {
 	return strings.Join(append([]string{"teleport"}, bucket...), "/")
 }
 
-// getRecords retrieve all prefixed keys
-func (b *DynamoDBBackend) getRecords(path string, opts ...backend.OpOption) ([]record, error) {
+type getResult struct {
+	records []record
+	// lastEvaluatedKey is the primary key of the item where the operation stopped, inclusive of the
+	// previous result set. Use this value to start a new operation, excluding this
+	// value in the new request.
+	lastEvaluatedKey map[string]*dynamodb.AttributeValue
+}
+
+func (b *DynamoDBBackend) getAllRecords(path string, opts ...backend.OpOption) ([]record, error) {
+	var result getResult
+	// this code is being extra careful here not to introduce endless loop
+	// by some unfortunate series of events
+	for i := 0; i < 100; i++ {
+		re, err := b.getRecords(path, result.lastEvaluatedKey, opts...)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		result.records = append(result.records, re.records...)
+		if len(re.lastEvaluatedKey) == 0 {
+			result.lastEvaluatedKey = nil
+			return result.records, nil
+		}
+		result.lastEvaluatedKey = re.lastEvaluatedKey
+	}
+	return nil, trace.BadParameter("backend entered endless loop")
```

Comment:
Have you considered returning whatever's been collected after 100 iterations instead of an error (and logging it too)?

Reply:
I think at this point it will be so much data, almost pointless